src/Type/Php/ClassImplementsFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\ObjectWithoutClassType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+use function count;
+use function in_array;
+
+class ClassImplementsFunctionReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array(
+			$functionReflection->getName(),
+			['class_implements', 'class_uses', 'class_parents'],
+			true,
+		);
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
+	{
+		if (count($functionCall->getArgs()) < 1) {
+			return null;
+		}
+
+		$objectOrClassType = $scope->getType($functionCall->getArgs()[0]->value);
+		$autoload = !isset($functionCall->getArgs()[1])
+			|| $scope->getType($functionCall->getArgs()[1]->value)->equals(new ConstantBooleanType(true));
+
+		if ($objectOrClassType instanceof UnionType) {
```

Comment:
No reason to do `instanceof *Type` here. A single `$this->canFunctionReturnFalse($objectOrClassType, $autoload)` call should be sufficient.

Reply:
If I just write
```
if ($this->canFunctionReturnFalse($objectOrClassType, $autoload)) {
     return null;
}
```
my tests are failing

Reply:
Nevermind, I found a way to do it.